### ISSUE TITLE
WebDAV sync does not work well

### ISSUE THREAD

**Issue by yarray** [HappenApps/Quiver-iOS#35]
Very impressive & promising beta! Hope it will be even better when released.

Two related issues:

1. Pull-to-refresh at the root will not get new notebooks.
2. Pull-to-refresh at the notebook will not update notes.

Currently I have to re-open the library and re-download the notebooks.

Btw, personally the start page after opening library is less relevant. Perhaps it's better to borrow some ideas from Evernote (with direct access to recent notes at the top, and notebook .etc beneath them)?

Well I am looking forward to flee from Evernote eventually...

**Comment by yarray** [comment]
A finding: When I click "+" on the right top, cancel, and re-open the note, the content will be refreshed. So I guess it is something about local cache?

Also I'm wondering is it possible to auto-refresh the note upon opening, or periodically. For seamless editing on multiple platforms this seems a must.

**Comment by mikenew12** [comment]
Just wanted to verify; same thing for me with WebDAV. Nothing refreshes unless I hit the '+' button and cancel.

There's also an additional issue that I'm sure is related. If I log in with my WebDAV account I can see the full directory structure and everything works fine (except the above issue obviously), but if I go to Settings -> Open an Existing Library and delete the account, nothing changes. I still have full access to everything. If I re-add the account with the same server address but put garbage for the username and password, it acts like it's logged in and still shows me all the data.

So I guess it either 1) never really disconnects once you connect to a server or 2) it caches the username and password and uses that every time to connect to the same server.

@ylian I do still have that test account active on my server with the username 'ylian' if you need it for any more debugging. There's really no immediate need for me to close it so feel free to use it anytime.

**Comment by ylian** [comment]
Two different issues here:

1. The note does not refresh unless you go to another note (or create a note) then come back. I will fix.
2. Deleting a WebDAV account doesn't clear the previous credential. I will look into it.

**Comment by ylian** [comment]
@mikenew12 Also regarding storing credentials, WebDAV credentials are saved into the iOS keychain, and every time a new request is made to the WebDAV server, the credential is used in the HTTP authorization header. WebDAV uses HTTP basic or digest authentication, so credentials are always sent with each request. There is no notion of a session, just like any regular HTTP request.

**Comment by mikenew12** [comment]
@ylian makes sense. Playing with it a bit more it looks like whatever you enter the first time is what it stores and uses. So if I enter my credentials incorrectly then it will fail every request, even if I go back and enter the correct credentials. So if you mistype your username or password there's no way to fix it unless you delete the app and re-install.

**Comment by mikenew12** [comment]
@ylian sorry, I spoke too soon. Trying it again I was able to get in after entering my credentials wrong.

**Comment by dgmorrisjr** [comment]
Confirmed that notes on IOS are not displaying the updated changes from the desktop app when using dropbox (webdav?)
